src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -149,6 +149,8 @@ sealed abstract class ArraySeq[+A]
     case _ =>
       super.equals(other)
   }
+
+  override def last: A = unsafeArray(unsafeArray.length - 1).asInstanceOf[A]
```

Comment:
what if the array is empty?

Reply:
current implementation (`IterableOps#last`) and my implementation both throw `ArrayIndexOutOfBoundsException`

```
Welcome to Scala 2.13.0-pre-6dfcda8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> val x = collection.immutable.ArraySeq[String]()
x: scala.collection.immutable.ArraySeq[String] = ArraySeq()

scala> x.last
java.lang.ArrayIndexOutOfBoundsException: 0
  at scala.collection.immutable.ArraySeq$ofRef.apply(ArraySeq.scala:227)
  at scala.collection.SeqView$Id.apply(SeqView.scala:24)
  at scala.collection.IndexedSeqView$IndexedSeqViewIterator.next(IndexedSeqView.scala:30)
  at scala.collection.IterableOps.last(Iterable.scala:210)
  at scala.collection.IterableOps.last$(Iterable.scala:208)
  at scala.collection.AbstractIterable.last(Iterable.scala:728)
  ... 38 elided
```

Reply:
Scala 2.12.6 `mutable.ArraySeq#last` throw `NoSuchElementException`

```
Welcome to Scala 2.12.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> val x = collection.mutable.ArraySeq[String]()
x: scala.collection.mutable.ArraySeq[String] = ArraySeq()

scala> x.last
java.util.NoSuchElementException: next on empty iterator
  at scala.collection.Iterator$$anon$2.next(Iterator.scala:38)
  at scala.collection.Iterator$$anon$2.next(Iterator.scala:36)
  at scala.collection.IndexedSeqLike$Elements.next(IndexedSeqLike.scala:60)
  at scala.collection.IterableLike.head(IterableLike.scala:106)
  at scala.collection.IterableLike.head$(IterableLike.scala:105)
  at scala.collection.mutable.ArraySeq.scala$collection$IndexedSeqOptimized$$super$head(ArraySeq.scala:46)
  at scala.collection.IndexedSeqOptimized.head(IndexedSeqOptimized.scala:125)
  at scala.collection.IndexedSeqOptimized.head$(IndexedSeqOptimized.scala:125)
  at scala.collection.mutable.ArraySeq.head(ArraySeq.scala:46)
  at scala.collection.TraversableLike.last(TraversableLike.scala:431)
  at scala.collection.TraversableLike.last$(TraversableLike.scala:430)
  at scala.collection.mutable.ArraySeq.scala$collection$IndexedSeqOptimized$$super$last(ArraySeq.scala:46)
  at scala.collection.IndexedSeqOptimized.last(IndexedSeqOptimized.scala:131)
  at scala.collection.IndexedSeqOptimized.last$(IndexedSeqOptimized.scala:131)
  at scala.collection.mutable.ArraySeq.last(ArraySeq.scala:46)
  ... 28 elided
```

Reply:
Did the migration guide include that?
I more like a 
```
if (array.length == 0) throw new NoSuchElementException else ...
```